packages/pkl.experimental.structuredRead/examples/envVars/envVars.pkl

```diff
@@ -0,0 +1,32 @@
+//===----------------------------------------------------------------------===//
+// Copyright © 2024 Apple Inc. and the Pkl project authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//===----------------------------------------------------------------------===//
+/// Example reading values from the enviroment to fill
+/// in a Pkl class.
+///
+/// Try running `pkl eval envVars.pkl`
+module envVars
+import ".../structuredRead.pkl"
+
+class EnvConfig {
+  SHELL: String?
+  USER: String?
+  PWD: String?
+  LANG: String?
+}
+
+output {
+  value = structuredRead.fromEnv.fill(EnvConfig)
```

Comment:
A neat little upside of having `fromFoo` _properties_ as opposed to modules is that you can now have _multiple_ and _with precedence_!

```
structuredRead.fromEnv.fromExternalProps.fill(EnvConfig)
```
would accept _either_ env vars, _or_ external properties, favouring the latter.

Reply:
Huh, that hadn't even occurred to me, I wonder what horrible abuses of this feature people will come up with 😄